public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx

```diff
@@ -0,0 +1,72 @@
+import * as H from 'history';
+
+import { locationService } from '@grafana/runtime';
+import { SceneObject, SceneObjectBase, SceneObjectState, sceneUtils, VizPanel } from '@grafana/scenes';
+
+import { DashboardScene } from '../scene/DashboardScene';
+import { getDashboardUrl } from '../utils/utils';
+
+import { PanelEditorRenderer } from './PanelEditorRenderer';
+
+export interface PanelEditorState extends SceneObjectState {
+  panel: VizPanel;
+  controls?: SceneObject[];
+  isDirty?: boolean;
```

Comment:
What do you think about exploring an alternative approach of having a higher level, object editor scene object that would try handling objects editing in a bit more generic way? Something like a wrapper around editable object.

would have an interface:

```
interface SceneObjectEditorState {
  isDirty: boolean;
  object: SceneObject;
}
```

think we gonna have land on something along these lines, otherwise each single object (variable, query, transformation) will have a custom edit implementation, while I have a gut feeling a lot of the change detection will be a shared implementation.

Reply:
Maybe for variable and annotation queries the UI and patterns are going to be similar. But for panel edit is going to be widely different (and it's its own route so also very different). the PanelEditor will contain a query editor, transform editor, viz panel editors etc. 

But maybe we can look for commonalities when we get there. Right now I cannot see how a generic scene object editor could work for panels and variables. etc. (Queries, and Transformations are part of panels).